File: Manual 10/Question03.cpp
#include <iostream>
using namespace std;

int main()
{
    double data[3][5]; // Array to store intake data for 3 monkeys over 5 days
    double least = 10000, greatest = -1, sum = 0, avg;

    // Input data for each monkey
    for (int i = 0; i < 3; i++)
    {
        cout << "\n\nFor Monkey#" << i + 1 << "\n\n";
        for (int j = 0; j < 5; j++)
        {
            cout << "Enter Amount for Day " << j + 1 << ": ";
            cin >> data[i][j];
        }
    }

    // Calculate sum, greatest, and least intake
    for (int i = 0; i < 3; i++)
    {
        for (int j = 0; j < 5; j++)
        {
            sum += data[i][j];
            if (data[i][j] > greatest)
            {
                greatest = data[i][j];
            }
            if (data[i][j] < least)
            {
                least = data[i][j];
            }
        }
    }

    // Calculate average intake
    avg = sum / 15;

    // Output results
    cout << "\n\nGreatest intake: " << greatest
         << "\nLeast Intake: " << least
         << "\nAverage Intake: " << avg << endl;

    return 0;
}

File: Python/Manual 10/Question03.py
def get_food_data():
    food_data = []
    for i in range(3):  # 3 monkeys
        week_data = []
        print(f"Enter the food data for monkey {i + 1}:")
        for j in range(5):  # 5 days
            while True:
                food = float(input(f"Day {j + 1}: "))
                if food < 0:
                    print("Invalid input, please enter a non-negative value.")
                else:
                    break
            week_data.append(food)
        food_data.append(week_data)
    return food_data


def report(food_data):
    average_food = sum(sum(week) for week in food_data) / (3 * 5)
    min_food = min(min(week) for week in food_data)
    max_food = max(max(week) for week in food_data)

    print(f"\nAverage food per day by all monkeys: {average_food:.2f}")
    print(
        f"Least amount of food eaten during the week by any one monkey: {min_food:.2f}"
    )
    print(
        f"Greatest amount of food eaten during the week by any one monkey: {max_food:.2f}"
    )


food_data = get_food_data()
report(food_data)
